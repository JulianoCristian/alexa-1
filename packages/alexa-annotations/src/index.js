export Skill from './Skill';
export annotation from './annotation';
export * as ErrorCode from './ErrorCodes';

/*******************************************************************************
 * Alexa Custom Skill Intents
 ******************************************************************************/
export Launch from './CustomSkill/Launch';
export Intent from './CustomSkill/Intent';
export SessionEnded from './CustomSkill/SessionEnded';

/*******************************************************************************
 * Alexa Smart Home Skill Intents
 ******************************************************************************/
export Discovery from './SmartHomeSkill/Discovery';
export Control from './SmartHomeSkill/Control';

export TurnOn from './SmartHomeSkill/TurnOn';
export TurnOff from './SmartHomeSkill/TurnOff';

export SetTargetTemperature from './SmartHomeSkill/SetTargetTemperature';
export IncrementTargetTemperature from './SmartHomeSkill/IncrementTargetTemperature';
export DecrementTargetTemperature from './SmartHomeSkill/DecrementTargetTemperature';

export SetPercentage from './SmartHomeSkill/SetPercentage';
export IncrementPercentage from './SmartHomeSkill/IncrementPercentage';
export DecrementPercentage from './SmartHomeSkill/DecrementPercentage';
