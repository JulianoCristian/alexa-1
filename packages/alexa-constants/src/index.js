export const Request = {
  ExceptionEncountered: 'System.ExceptionEncountered',
  Intent: 'IntentRequest',
  Launch: 'LaunchRequest',
  SessionEnded: 'SessionEndedRequest'
};

export const Intent = {
  Cancel: 'AMAZON.CancelIntent',
  Help: 'AMAZON.HelpIntent',
  LoopOff: 'AMAZON.LoopOffIntent',
  LoopOn: 'AMAZON.LoopOnIntent',
  Next: 'AMAZON.NextIntent',
  No: 'AMAZON.NoIntent',
  Pause: 'AMAZON.PauseIntent',
  Previous: 'AMAZON.PreviousIntent',
  Repeat: 'AMAZON.RepeatIntent',
  Resume: 'AMAZON.ResumeIntent',
  ShuffleOff: 'AMAZON.ShuffleOffIntent',
  ShuffleOn: 'AMAZON.ShuffleOnIntent',
  StartOver: 'AMAZON.StartOverIntent',
  Stop: 'AMAZON.StopIntent',
  Yes: 'AMAZON.YesIntent'
};

export const AudioPlayer = {
  Request: {
    PlaybackFailed: 'AudioPlayer.PlaybackFailed',
    PlaybackFinished: 'AudioPlayer.PlaybackFinished',
    PlaybackNearlyFinished: 'AudioPlayer.PlaybackNearlyFinished',
    PlaybackStarted: 'AudioPlayer.PlaybackStarted',
    PlaybackStopped: 'AudioPlayer.PlaybackStopped'
  },

  Directive: {
    ClearQueue: 'AudioPlayer.ClearQueue',
    Play: 'AudioPlayer.Play',
    Stop: 'AudioPlayer.Stop'
  }
};

export const PlaybackController = {
  Request: {
    NextCommandIssued: 'PlaybackController.NextCommandIssued',
    PauseCommandIssued: 'PlaybackController.PauseCommandIssued',
    PlayCommandIssued: 'PlaybackController.PlayCommandIssued',
    PreviousCommandIssued: 'PlaybackController.PreviousCommandIssued'
  }
};

export const ConnectedHome = {
  Control: 'Alexa.ConnectedHome.Control',
  Discovery: 'Alexa.ConnectedHome.Discovery',

  Request: {
    DecrementPercentage: 'DecrementPercentageRequest',
    DecrementTargetTemperature: 'DecrementTargetTemperatureRequest',
    DiscoverAppliances: 'DiscoverAppliancesRequest',
    IncrementPercentage: 'IncrementPercentageRequest',
    IncrementTargetTemperature: 'IncrementTargetTemperatureRequest',
    SetPercentage: 'SetPercentageRequest',
    SetTargetTemperature: 'SetTargetTemperatureRequest',
    TurnOff: 'TurnOffRequest',
    TurnOn: 'TurnOnRequest'
  },

  Confirmation: {
    DecrementPercentage: 'DecrementPercentageConfirmation',
    DecrementTargetTemperature: 'DecrementTargetTemperatureConfirmation',
    IncrementPercentage: 'IncrementPercentageConfirmation',
    IncrementTargetTemperature: 'IncrementTargetTemperatureConfirmation',
    SetPercentage: 'SetPercentageConfirmation',
    SetTargetTemperature: 'SetTargetTemperatureConfirmation',
    TurnOff: 'TurnOffConfirmation',
    TurnOn: 'TurnOnConfirmation'
  }
};

export default {
  AudioPlayer,
  ConnectedHome,
  Intent,
  PlaybackController,
  Request
};
