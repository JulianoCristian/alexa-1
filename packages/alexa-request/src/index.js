export default from './Request';

// For commonjs compatibility
export const session = Request.session;
export const intent = Request.intent;
export const launchRequest = Request.launchRequest;
export const sessionEndedRequest = Request.sessionEndedRequest;
