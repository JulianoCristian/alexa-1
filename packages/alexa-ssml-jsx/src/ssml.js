import schema from './schema/index';
import validate from './prop-types/validate';

export default function ssml(tagName, props, ...children) {
  const tag = throwIfUndefined(getTagDefinition(tagName), `Unsupported tag: "${tagName}"`);

  const hasProps = props || tag.defaultProps || children.length;
  const mergedProps = hasProps && { ...tag.defaultProps, ...props, ...(children.length && { children }) };
  const validatedProps = validate(tag, mergedProps);

  return { type: tag.type, ...(validatedProps && { props: validatedProps }) };
};

function throwIfUndefined(item, error) {
  if (!item) {
    throw new Error(error);
  } else {
    return item;
  }
}

function getTagDefinition(tag) {
  switch (typeof tag) {
  case 'string':
    return schema[tag];
  case 'function':
    return { ...tag, type: tag };
  default:
    return undefined;
  }
}
