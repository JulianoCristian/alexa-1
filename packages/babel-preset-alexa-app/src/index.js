const presets = [
  // Latest stable ECMAScript features available on AWS Lambda
  [
    require.resolve('babel-preset-env'),
    {
      targets: {
        node: '6.10'
      }
    }
  ],

  // Flow
  require.resolve('babel-preset-flow')
];

const plugins = [
  // class { handleClick = () => { } }
  require.resolve('babel-plugin-transform-class-properties'),

  // { ...todo, completed: true }
  [
    require.resolve('babel-plugin-transform-object-rest-spread'),
    {
      useBuiltIns: true
    },
  ],

  // Transforms SSML
  [
    require.resolve('babel-plugin-transform-jsx'),
    {
      'module': 'ssml-jsx',
      'function': 'ssml',
      'useVariables': true
    }
  ],

  // Polyfills the runtime needed for async/await and generators
  [
    require.resolve('babel-plugin-transform-runtime'),
    {
      helpers: false,
      polyfill: false,
      regenerator: true
    },
  ],

  // function* () { yield 42; yield 43; }
  [
    require.resolve('babel-plugin-transform-regenerator'),
    {
      // Async functions are converted to generators by babel-preset-env
      async: false
    },
  ],

  // Adds syntax support for import()
  require.resolve('babel-plugin-syntax-dynamic-import'),

  // Add support for decorators
  require.resolve('babel-plugin-transform-decorators-legacy'),

  // Convert exports to commonjs
  require.resolve('babel-plugin-add-module-exports')
];

// This is similar to how `env` works in Babel:
// https://babeljs.io/docs/usage/babelrc/#env-option
// We are not using `env` because it’s ignored in versions > babel-core@6.10.4:
// https://github.com/babel/babel/issues/4539
// https://github.com/facebookincubator/create-react-app/issues/720
// It’s also nice that we can enforce `NODE_ENV` being specified.
var env = process.env.BABEL_ENV || process.env.NODE_ENV;
if (env !== 'development' && env !== 'test' && env !== 'production') {
  throw new Error(
    'Using `babel-preset-alexa-app` requires that you specify `NODE_ENV` or ' +
      '`BABEL_ENV` environment variables. Valid values are "development", ' +
      '"test", and "production". Instead, received: ' +
      JSON.stringify(env) +
      '.'
  );
}

module.exports = {
  presets,
  plugins
};
