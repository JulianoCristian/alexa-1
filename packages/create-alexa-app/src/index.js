const fs = require('fs-extra');
const path = require('path');
const run = require('./run');
const validateName = require('./validateName');

const chain = (...promises) => promises.reduce((state, action) => state.then(action), Promise.resolve());

const error = (name) => (err) => Promise.reject({ name, error: err });

const preinstall = (projectDirectory) => () => {
  const root = path.resolve(projectDirectory);
  const appName = path.basename(root);
  return validateName(appName)
    .then(() => ({ appName, root }))
    .catch(error('Pre-install'));
};

const install = (packageJson) => ({ appName, root }) => {
  const packagePath = path.join(root, 'package.json');
  return fs
    .ensureDir(root)
    .then(() => fs.writeJson(packagePath, Object.assign({ name: appName }, packageJson), { spaces: 2 }))
    .catch(error('Package creation'))
    .then(() => process.chdir(root))
    .then(() => run('npm', 'install'))
    .catch(error('Dependency installation'))
    .then(() => run('npm', 'run', 'init', '--', root))
    .catch(error('Project initialization'))
    .then(() => root);
};

const postinstall = (root) => {
  const packageJson = require(path.join(root, 'package.json'));
  const scripts = Object.assign({}, packageJson.scripts, { init: undefined });
  const newPackageJson = Object.assign({}, packageJson, { scripts });
  return fs.writeJson(path.join(root, 'package.json'), newPackageJson, { spaces: 2 }).catch(error('Post-install'));
};

module.exports = (projectDirectory, configuration) =>
  chain(preinstall(projectDirectory), install(configuration.package.json), postinstall).catch((error) => {
    // Delete known generated files
    return Promise.reject(error);
  });
