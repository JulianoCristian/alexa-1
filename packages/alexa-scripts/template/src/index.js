import { Skill, Launch, Intent } from 'alexa-annotations';
import Response, { say, ask } from 'alexa-response';

export class HelloWorld {
  @Launch
  launch() {
    return Response.build({
      ask: 'Hello World Alexa skill launched! Who would you like to say hello to?',
      reprompt: 'Who would you like to say hello to?'
    });
  }

  @Intent('hello')
  hello({ name = 'world' }) {
    return Response.build({
      say: <speak>Hello {name}!</speak>,
      card: {
        title: 'Hello toâ€¦',
        content: `Hello ${name}!`
      }
    });
  }

  @Intent('AMAZON.HelpIntent')
  help() {
    return ask('I say hello to people. Who should I say hello to?').reprompt('Who should I say hello to?');
  }

  @Intent('AMAZON.CancelIntent', 'AMAZON.StopIntent')
  stop() {
    return say(<speak>Goodbye!</speak>);
  }
}

export default Skill(HelloWorld);
