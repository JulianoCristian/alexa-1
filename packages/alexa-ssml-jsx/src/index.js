import ssml, { renderToString as rts } from 'ssml-jsx';

export default ssml;
export const renderToString = rts;
