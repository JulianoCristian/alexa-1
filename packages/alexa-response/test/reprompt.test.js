import { reprompt } from '../src';
import ssml from 'ssml-jsx';

test('PlainText', () => {
  expect(reprompt('Can you repeat that?').build()).toMatchSnapshot();
});

test('SSML', () => {
  expect(
    reprompt(<speak>Can you repeat that?</speak>).build()
  ).toMatchSnapshot();
});
