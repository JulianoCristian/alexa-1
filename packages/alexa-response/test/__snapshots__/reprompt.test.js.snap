// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainText 1`] = `
Object {
  "response": Object {
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "Can you repeat that?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;

exports[`SSML 1`] = `
Object {
  "response": Object {
    "reprompt": Object {
      "outputSpeech": Object {
        "ssml": "<speak>Can you repeat that?</speak>",
        "type": "SSML",
      },
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;
